src/tests/functional/shared_test_classes/src/single_layer/generate_proposals.cpp

```diff
@@ -110,6 +121,54 @@ void GenerateProposalsLayerTest::generate_inputs(const std::vector<ngraph::Shape
     }
 }
 
+void GenerateProposalsLayerTest::compare(const std::vector<ov::Tensor>& expected,
```

Comment:
Do we need specific compare() for the GenerateProposal?

Reply:
Yes, we do. The thing is that reference GenerateProposal implementation produces output where **num_rois** value is dynamic. GPU plugin does not support dynamic shapes, so its output shape is static with the maximum possible value in 1st dimension - **num_batches** * **post_nms_count**. 1st **num_rois** records must match reference output, while the rest are filled with zeroes. That's what custom compare() checks.